pyccel/parser/syntactic.py

```diff
@@ -796,20 +805,31 @@ def fill_types(ls):
 
             results.append(result_name)
 
-        func = FunctionDef(
-               name,
-               arguments,
-               results,
-               body,
-               local_vars=local_vars,
-               global_vars=global_vars,
-               is_pure=is_pure,
-               is_elemental=is_elemental,
-               is_private=is_private,
-               imports=imports,
-               decorators=decorators,
-               headers=headers,
-               doc_string=doc_string)
+        kwargs = {
+               'local_vars':local_vars,
+               'global_vars':global_vars,
+               'is_pure':is_pure,
+               'is_elemental':is_elemental,
+               'is_private':is_private,
+               'imports':imports,
+               'decorators':decorators,
+               'headers':headers,
+               'doc_string':doc_string
+               }
+        if is_inline:
+            func = InlineFunctionDef(
+                   name,
+                   arguments,
+                   results,
+                   body,
+                   **kwargs)
+        else:
+            func = FunctionDef(
+                   name,
+                   arguments,
+                   results,
+                   body,
+                   **kwargs)
 
         return func
```

Comment:
Could we write all of this in two lines?
```suggestion
        cls = InlineFunctionDef if is_inline else FunctionDef
        return cls(name, arguments, results, body, **kwargs)
```

Reply:
That's much better. I thought something like this must exist which didn't require kwargs I just couldn't work out what it was. I'm not used to saving types.

Reply:
(+ correction e1ed594)